internal/instance/resource_instance.go

```diff
@@ -1359,3 +1405,29 @@ func findIPv6Address(network api.InstanceStateNetwork) (string, bool) {
 
 	return ipv6, (ipv6 != "")
 }
+
+// Find all global IPv6 addresses or return any IPv6 address
+// if there are no global addresses. This works analog to the IPv4
+// selection mechanism but favours global addresses.
+func findIPv6Addresses(network api.InstanceStateNetwork) ([]string, bool) {
+	var ipv6 []string
+
+	for _, ip := range network.Addresses {
+		if ip.Family == "inet6" && ip.Scope == "global" {
+			ipv6 = append(ipv6, ip.Address)
+			// ipv6 = ip.Address
```

Comment:
Left over comment?

Reply:
and this is why i'm not supposed to make PRs late at night :sweat_smile:

thanks